libs/community/langchain_community/vectorstores/lancedb.py

```diff
@@ -52,25 +57,41 @@ def __init__(
                 "Could not import lancedb python package. "
                 "Please install it with `pip install lancedb`."
             )
-        self.lancedb = lancedb
         self._embedding = embedding
         self._vector_key = vector_key
         self._id_key = id_key
         self._text_key = text_key
         self._table_name = table_name
+        self.api_key = api_key
+        self.region = region
+        self.mode = mode
+
+        if os.getenv("LANCE_API_KEY") is not None:
+            api_key = os.getenv("LANCE_API_KEY")
```

Comment:
Here we need to check for either self.api_key or ENV var right?

```
self.api_key = api_key or os.getenv("Lance_api_key")
```

Reply:
Yes added that in if else can condense in one line, thanks

Reply:
but then here self.api_key becomes useless?